Automattic/jetpack
# PHP Compatibility: fix dynamic property deprecation notices

## Proposed changes:

Follow-up to #27949. Fix more deprecation notices introduced with PHP 8.2:

```
PHP Deprecated:  Creation of dynamic property Automattic\Jetpack\Search\Customizer::$plan is deprecated in /wp-content/plugins/jetpack-dev/jetpack_vendor/automattic/jetpack-search/src/customizer/class-customizer.php on line 24
```

This was reported here:
https://wordpress.org/support/topic/dynamic-property/

### Other information:


## Jetpack product discussion

Primary issue: #27927

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Load this branch on a site running PHP 8.2, and a site running an old version of PHP.
* Ensure each feature touched by this PR still works.
